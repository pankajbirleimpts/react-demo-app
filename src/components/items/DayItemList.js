import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { Grid, Checkbox } from "semantic-ui-react";
import { getAllDayItems } from "../../actions/ItemAction";
import { Loader, CustomTable } from "../common";

class DayItemList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      columns: [
        {
          label: "Item Name",
          keyName: "itemName",
          sort: true,
          searchable: true,
          sortable: true
        },
        {
          label: "Category",
          keyName: "category",
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: "Date",
          keyName: "date",
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: "Amount",
          keyName: "amount",
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: "Total Quantity",
          keyName: "totalQuantity",
          sort: null,
          searchable: true,
          sortable: true
        },
        {
          label: "Sell Out Quantity",
          keyName: "remainingQuantity",
          sort: null,
          searchable: true,
          sortable: true
        }
      ]
    };
  }

  componentDidMount() {
    this.props.getAllDayItems();
  }

  
  render() {
    return (
      <Grid>
        <Loader isLoading={this.props.item.isLoading} />
        <Grid.Row>
          <Grid.Column width={8} textAlign="left"><h3>Manage Day Items</h3></Grid.Column>
          <Grid.Column width={8} textAlign="right"><Link to="/add-day-item" className="ui button primary">+ Add Day item</Link></Grid.Column>
        </Grid.Row>
        <Grid.Row centered>
          <CustomTable
            columns={this.state.columns}
            tableData={this.props.item.allDayItems}
            module="DAYITEMS"
          />
        </Grid.Row>
      </Grid>
    );
  }
}

function mapStateToProp({ item }) {
  return {
    item
  };
}

export default connect(mapStateToProp, { getAllDayItems })(DayItemList);
