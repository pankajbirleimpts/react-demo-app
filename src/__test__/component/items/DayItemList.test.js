import React from "react";
import { shallow } from "enzyme";
import DayItemList from "../../../components/items/DayItemList";
import { storeFactory } from "../../test/TestUtils";

const setup = props => {
  const store = storeFactory(props);
  const wrapper = shallow(<DayItemList store={store} getAllDayItems={jest.fn()}/>)
    .dive()
    .dive();
  return wrapper;
};

describe("DayItemList component", () => {
  const props = {
    item: {
      isLoading: false,
      allDayItems: [
        {
          id:1,
          date: "27-Dec-2019",
          remainingQuantity: 0,
          totalQuantity: 100,
          itemName: "Dal tadka",
          category: "Dal",
          amount: 10
        },
        {
          id:2,
          date: "27-Dec-2019",
          remainingQuantity: 0,
          totalQuantity: 100,
          itemName: "Sev Tamatar",
          category: "Dal",
          amount: 20
        }
      ]
    }
  };
  test("Snapshot of DayItemList component", () => {
    const wrapper = setup(props);
    expect(wrapper).toMatchSnapshot();
  });

  test("Check the isLoading in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.item.isLoading).toBe(props.item.isLoading);
  });

  test("Check the `allDayItems` in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.item.allDayItems).toBe(props.item.allDayItems);
  });

  test("Check the `getAllDayItems` action in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.getAllDayItems).toBeInstanceOf(Function);
  });
});
