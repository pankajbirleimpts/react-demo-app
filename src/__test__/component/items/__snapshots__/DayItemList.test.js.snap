// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayItemList component Snapshot of DayItemList component 1`] = `
<Grid>
  <Loading
    isLoading={false}
  />
  <GridRow>
    <GridColumn
      textAlign="left"
      width={8}
    >
      <h3>
        Manage Day Items
      </h3>
    </GridColumn>
    <GridColumn
      textAlign="right"
      width={8}
    >
      <Link
        className="ui button primary"
        to="/add-day-item"
      >
        + Add Day item
      </Link>
    </GridColumn>
  </GridRow>
  <GridRow
    centered={true}
  >
    <Connect(UnconnectedCustomTable)
      columns={
        Array [
          Object {
            "keyName": "itemName",
            "label": "Item Name",
            "searchable": true,
            "sort": true,
            "sortable": true,
          },
          Object {
            "keyName": "category",
            "label": "Category",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "date",
            "label": "Date",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "amount",
            "label": "Amount",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "totalQuantity",
            "label": "Total Quantity",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "remainingQuantity",
            "label": "Sell Out Quantity",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
        ]
      }
      module="DAYITEMS"
      tableData={
        Array [
          Object {
            "amount": 10,
            "category": "Dal",
            "date": "27-Dec-2019",
            "id": 1,
            "itemName": "Dal tadka",
            "remainingQuantity": 0,
            "totalQuantity": 100,
          },
          Object {
            "amount": 20,
            "category": "Dal",
            "date": "27-Dec-2019",
            "id": 2,
            "itemName": "Sev Tamatar",
            "remainingQuantity": 0,
            "totalQuantity": 100,
          },
        ]
      }
    />
  </GridRow>
</Grid>
`;
